arrabiata/src/constraints.rs

```diff
@@ -0,0 +1,359 @@
+use super::{columns::Column, interpreter::InterpreterEnv};
+use crate::{
+    columns::{Gadget, E},
+    interpreter::Side,
+    MAX_DEGREE, NUMBER_OF_COLUMNS, NUMBER_OF_PUBLIC_INPUTS,
+};
+use ark_ff::{Field, PrimeField};
+use kimchi::circuits::{
+    expr::{ConstantTerm::Literal, Expr, ExprInner, Operations, Variable},
+    gate::CurrOrNext,
+};
+use log::debug;
+use num_bigint::BigInt;
+use o1_utils::FieldHelpers;
+
+pub struct Env<Fp: Field> {
+    pub poseidon_mds: Vec<Vec<Fp>>,
+    /// The parameter a is the coefficients of the elliptic curve in affine
+    /// coordinates.
+    // FIXME: this is ugly. Let use the curve as a parameter. Only lazy for now.
+    pub a: BigInt,
```

Comment:
I don't see that using a big integer is better than a field element

Reply:
We should bench later.